plugins/catalog-backend/src/service/CatalogBuilder.ts

```diff
@@ -465,7 +465,7 @@ export class CatalogBuilder {
     const locationAnalyzer =
       this.locationAnalyzer ?? new RepoLocationAnalyzer(logger, integrations);
     const locationService = new AuthorizedLocationService(
-      new DefaultLocationService(locationStore, orchestrator),
+      new DefaultLocationService(locationStore, orchestrator, config),
```

Comment:
Thinking that we should have a method on here which would take the new types.

`builder.setAllowedLocationTypes(['url', 'mycustomtpye'])` or something like that.